mycroft/skills/core.py

```diff
@@ -591,45 +590,21 @@ def wrapper(message):
 
                 stopwatch = Stopwatch()
                 with stopwatch:
-                    if need_self:
-                        # When registring from decorator self is required
-                        if len(getargspec(handler).args) == 2:
-                            handler(self, unmunge_message(message,
-                                                          self.skill_id))
-                        elif len(getargspec(handler).args) == 1:
-                            handler(self)
-                        elif len(getargspec(handler).args) == 0:
-                            # Zero may indicate multiple decorators, trying the
-                            # usual call signatures
-                            try:
-                                handler(self, unmunge_message(message,
-                                                              self.skill_id))
-                            except TypeError:
-                                handler(self)
-                        else:
-                            LOG.error("Unexpected argument count:" +
-                                      str(len(getargspec(handler).args)))
-                            raise TypeError
+                    is_bound = bool(getattr(handler, 'im_self', None))
```

Comment:
May possibly be simplified slightly by using 
`is_bound = hasattr('im_self')`

Reply:
From what I understand, if a class method is given accessed statically via `ClassName.function`, `im_self` will exists but be `None`.

Reply:
Ah, right you are. Would fail for classmethods